samples/speech_transcribe_async.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2019, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+function main(
+  filepath = 'path/to/audio.raw',
+  encoding = 'LINEAR16',
+  sampleRateHertz = 16000,
+  languageCode = 'en-US'
+) {
+  // [START speech_transcribe_async]
+  /**
+   * TODO(developer): Uncomment these variables before running the sample.
+   */
+  // const filepath = 'path/to/audio.raw';
+  // const encoding = 'LINEAR16';
+  // const sampleRateHertz = 16000;
+  // const languageCode = 'en-US';
+
+  // Imports the Google Cloud Speech API client library.
+  const speech = require('@google-cloud/speech');
+
+  // Import other required libraries.
+  const fs = require('fs');
+  const util = require('util');
+  const readFile = util.promisify(fs.readFile);
+
+  // Creates a client.
+  const client = new speech.SpeechClient();
+
+  async function transcribeAsync() {
```

Comment:
Why does this function take no parameters?

As a JavaScript developer, would it not be common for me to read online documentation of functions with standard `@param name {type} description`

e.g. [example](https://github.com/googleapis/gapic-generator/blob/048de1a7dfd88f627b7bdab13cf608f6badf06cf/src/test/java/com/google/api/codegen/gapic/testdata/nodejs/nodejs_library.baseline#L459-L471) from generated sample

Reply:
I don't believe the snippet on cloud site will include the function, right?  My assumption was that we're very much optimizing these for consumption from cloud site, not the repo.

Reply:
The snippet on cloud site does include the function.

1. We must to show the function for Node.
  - Otherwise we use `await` but the snippet does not show the outer `async` function context (we've had bug reports from users running our snippets by copy/pasting and then Node.js blows up because we're using `await` but they didn't copy/paste a wrapping `async` function.
2. All languages** have standardized on showing method definitions in the embedded snippets. Especially important for languages where an `import` must happen above/outside of a method, so that we can show the import and the sample. Also has been very useful because it gives us the ability to use standard `@param` descriptions of each input parameter

** all but 1 or 2 which have not made clear decisions one way or the other afaict

> `#misc` I have standardized on using (a) methods/functions (b) standard parameter documentation across all languages for the first version of the generator. It's very consistent. It looks good. Every developer is accustomed to reading the format. It includes the **type** of each param. In the near future we can consider changing this standard if absolutely necessary, but it works really, really, really well.

Reply:
For this sample, the user on cloud.google.com will see exactly:

```js
/**
 * TODO(developer): Uncomment these variables before running the sample.
 */
// const filepath = 'path/to/audio.raw';
// const encoding = 'LINEAR16';
// const sampleRateHertz = 16000;
// const languageCode = 'en-US';

// Imports the Google Cloud Speech API client library.
const {SpeechClient} = require('@google-cloud/speech');

// Import other required libraries.
const fs = require('fs');
const util = require('util');
const readFile = util.promisify(fs.readFile);

// Creates a client.
const speech = new SpeechClient();

async function transcribeAsync() {
  // Get the bytes of the file.
  const fileBytes = Buffer.from(await readFile(filepath)).toString('base64');

  const request = {
    audio: {
      content: fileBytes,
    },
    config: {
      // Encoding of the audio file, e.g. 'LINEAR16'.
      encoding: encoding,
      // Sample rate in Hertz of the audio data sent.
      sampleRateHertz: sampleRateHertz,
      // BCP-47 language code, e.g. 'en-US'.
      languageCode: languageCode,
    },
  };

  // Detects speech in the audio file using a long-running operation. You can
  // wait for now, or get its results later.
  const [operation] = await speech.longRunningRecognize(request);

  // Wait for operation to complete.
  const [response] = await operation.promise();
  const transcriptions = response.results;
  console.log(`Transcriptions: ${transcriptions.length}`);
  for (const transcription of transcriptions) {
    console.log(transcription.alternatives[0].transcript);
  }
}

transcribeAsync();
```

1. The input parameters are commented out at the top.
2. The visible function is only there to enable `async`/`await`
3. The visible function is invoked at the very bottom of the sample
4. There's no point passing the commented out variables to the function because they're already in scope and visible from within the function. Adding them to the function signature and the function invocation would clutter the sample.
5. The sample is still runnable from the command-line by someone who cloned the repo because outside the region tags there is an additional wrapping function that _does_ take all the input parameters received from the command-line.